mapfishapp/src/main/webapp/app/js/GEOR_mapinit.js

```diff
@@ -313,11 +313,23 @@ GEOR.mapinit = (function() {
         var records = [], record;
         var errors = [], count = 0;
         Ext.each(initState, function(item) {
-            if (item.type == "WMSLayer" || item.type == "WFSLayer") {
+            if ( (item.type == "WMSLayer" || item.type == "WFSLayer") && item.type == type+'Layer' ) {
                 record = stores[item.url].queryBy(function(r) {
                     return (r.get('name') == item.name);
                 }).first();
                 if (record) {
+                    // handle cql_filter param in JSON POST
+                    if( type == "WFS" ) {
+                        if ( item.hasOwnProperty("cql_filter") ) {
+                            var format = new OpenLayers.Format.CQL();
```

Comment:
you probably miss the build dependency to OpenLayers/Format/CQL.js in the file header too